dev/Repeater/UniformGridLayoutState.cpp

```diff
@@ -42,24 +42,46 @@ void UniformGridLayoutState::EnsureElementSize(
     if (context.ItemCount() > 0)
     {
         // If the first element is realized we don't need to get it from the context
-        if (auto realizedElement = m_flowAlgorithm.GetElementIfRealized(0))
+        if (auto realizedElement = m_flowAlgorithm.GetElementIfRealized(0).as<winrt::FrameworkElement>())
         {
+            SetConstraints(realizedElement, availableSize, orientation, layoutItemWidth, layoutItemHeight);
             realizedElement.Measure(availableSize);
-            SetSize(realizedElement.DesiredSize(), layoutItemWidth, LayoutItemHeight, availableSize, stretch, orientation, minRowSpacing, minColumnSpacing, maxItemsPerLine);
+            SetSize(realizedElement.DesiredSize(), layoutItemWidth, layoutItemHeight, availableSize, stretch, orientation, minRowSpacing, minColumnSpacing, maxItemsPerLine);
         }
         else
         {
             // Not realized by flowlayout, so do this now!
-            if (const auto firstElement = context.GetOrCreateElementAt(0, winrt::ElementRealizationOptions::ForceCreate))
+            if (const auto firstElement = context.GetOrCreateElementAt(0, winrt::ElementRealizationOptions::ForceCreate).as<winrt::FrameworkElement>())
             {
+                SetConstraints(firstElement, availableSize, orientation, layoutItemWidth, layoutItemHeight);
                 firstElement.Measure(availableSize);
-                SetSize(firstElement.DesiredSize(), layoutItemWidth, LayoutItemHeight, availableSize, stretch, orientation, minRowSpacing, minColumnSpacing, maxItemsPerLine);
+                SetSize(firstElement.DesiredSize(), layoutItemWidth, layoutItemHeight, availableSize, stretch, orientation, minRowSpacing, minColumnSpacing, maxItemsPerLine);
                 context.RecycleElement(firstElement);
             }
         }
     }
 }
 
+void UniformGridLayoutState::SetConstraints(const winrt::UIElement element,
+    const winrt::Size availableSize,
+    const winrt::Orientation orientation,
+    const double itemWidth,
+    const double itemHeight)
+{
+    if (const auto frElement = element.try_as<winrt::FrameworkElement>()) {
+        if (orientation == winrt::Orientation::Horizontal) {
+            if (!isnan(itemWidth)) {
+                frElement.MaxWidth(itemWidth);
```

Comment:
Please correct me if I'm wrong, but is this essentially setting `MaxWidth` to `UniformGridLayout.MinItemWidth`? If that's the case, I think it would be best to borrow code from `SetSize()` below to get the effective size based on `UniformGridLayout.ItemsStretch`, as the effective size won't always match `MinItemWidth` and `MinItemHeight`.

Additionally, I wonder if setting `MaxWidth` and `MaxHeight` would interfere with cases where the user has already set those values in their `ItemTemplate`. Perhaps instead of setting `MaxWidth` and `MaxHeight`, this method could return a `Size` which would be used to measure `realizedElement` in `EnsureElementSize()`.

Reply:
Good idea, changed the code to the corresponding `winrt::Size` object to the elements `Measure`.

Reply:
I see, that's a good change, but unfortunately I'm not sure if it will fix the issue when `UniformGridLayout.ItemsStretch = Fill`, as the width of a column won't be equal to `UniformGridLayout.MinItemWidth`. We need to calculate the actual width of a column, and then use that value to measure the element.

For example, if `UniformGridLayout.MinItemWidth = 250`, and the `ItemsRepeater` is in a 1200px width container, I think the logic to determine the column width would go something like this:
* Divide container width by `MinItemWidth` to get column count: 1200 / 250 = 4
* Divide container width by column count to get effect width: 1200 / 4 = 300
* Measure first element using `new Size(300, availableWidth.Height)`

You would have columns that are 300px wide, not 250px wide, so measuring them with `MinItemWidth` alone would create incorrect results. `MinColumnSpacing` and `MinRowSpacing` would have to be accounted for as well.

I'm happy help with this if you'd like :)

Reply:
Very valid point. I updated the code now so that we take stretching into account. Please let me know your thoughts @Ryken100! Thank you for helping me with this PR :)